src/dev-spaces-preview/azext_dev_spaces_preview/custom.py

```diff
@@ -37,37 +37,28 @@ def ads_use_dev_spaces(cluster_name, resource_group_name, space_name='default',
     azds_cli = _install_dev_spaces_cli()
 
     from subprocess import PIPE
-    should_create_resource = False
+    resource_ready_code = 0  # Flag if the resource is ready, after create or select
     retCode = subprocess.call(
         [azds_cli, 'resource', 'select', '-n', cluster_name, '-g', resource_group_name],
         stderr=PIPE)
     if retCode == 1:
-        should_create_resource = True
-
-    if should_create_resource:
-        retCode = subprocess.call(
+        resource_ready_code = subprocess.call(
             [azds_cli, 'resource', 'create', '--aks-name', cluster_name, '--aks-resource-group',
              resource_group_name, '--name', cluster_name, '--resource-group', resource_group_name],
             universal_newlines=True)
 
-        if retCode == 0:
-            should_create_spaces = False
-            create_space_arguments = [azds_cli, 'space', 'select', '--name', space_name]
+    if resource_ready_code == 0:
```

Comment:
Just `return` if `resource_ready_code != 0`. This will remove the need to have the large amount of code under this if statement and the need for initializing the variable in line 40.